common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/AkkaContainerClient.scala

```diff
@@ -135,13 +136,17 @@ protected class AkkaContainerClient(
                               timeout: FiniteDuration,
                               retry: Boolean,
                               retryCount: Int = 0): Future[(HttpResponse, Int)] = {
+    val start = Instant.now
+
     request(req)
       .map((_, retryCount))
       .recoverWith {
         case t: StreamTcpException if retry =>
-          val newTimeout = timeout - retryInterval
-          if (newTimeout > Duration.Zero) {
-            akka.pattern.after(retryInterval, as.scheduler)(retryingRequest(req, newTimeout, retry, retryCount + 1))
+          if (timeout > Duration.Zero) {
+            akka.pattern.after(retryInterval, as.scheduler)({
+              val newTimeout = timeout - (Instant.now.toEpochMilli - start.toEpochMilli).milliseconds
```

Comment:
For future reference: you should be able to do this in a typesafe way by importing implicit conversions for durations (https://github.com/apache/incubator-openwhisk/blob/master/common/scala/src/main/scala/org/apache/openwhisk/utils/TimeHelpers.scala) and then do something along the lines of:

```scala
timeout - Duration.between(Instant.now, start)
```

to avoid going into Long and back.